api/v1/platformnetwork_types.go

```diff
@@ -7,62 +7,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// AllocationRange defines the start and end address for an allocation range
-type AllocationRange struct {
-	// Start defines the beginning of the address range (inclusively)
-	Start string `json:"start"`
-
-	// End defines the end of the address range (inclusively)
-	End string `json:"end"`
-}
-
-// AllocationInfo defines the allocation scheme details for a network
-type AllocationInfo struct {
-	// Ranges defines the pools from which host addresses are allocated.   If
-	// omitted addresses the entire network
-	// address space is considered available.
-	Ranges []AllocationRange `json:"ranges,omitempty"`
-
-	// Type defines whether network addresses are allocated dynamically or
-	// statically.
-	// +kubebuilder:validation:Enum=static;dynamic
-	Type string `json:"type"`
-
-	// Order defines whether host address are allocation randomly or sequential
-	// from the available pool or addresses.
-	// +kubebuilder:validation:Enum=sequential;random
-	// +optional
-	Order *string `json:"order,omitempty"`
-}
-
 // PlatformNetworkSpec defines the desired state of PlatformNetwork
 type PlatformNetworkSpec struct {
 	// Type defines the intended usage of the network
 	// +kubebuilder:validation:Enum=mgmt;pxeboot;infra;oam;multicast;system-controller;cluster-host;cluster-pod;cluster-service;storage;admin;other
 	Type string `json:"type"`
 
-	// Subnet defines the subdivision IPv4 or IPv6 network address for the network
-	Subnet string `json:"subnet"`
-
-	// FloatingAddress defines the floating IPv4 or IPv6 network address for the network
-	FloatingAddress string `json:"floatingAddress,omitempty"`
-
-	// Controller0Address is the controller-0 IPv4 or IPv6 network address value.
-	Controller0Address string `json:"controller0Address,omitempty"`
+	// Dynamic defines whether network addresses are allocated dynamically (true) or
+	// statically (false).
+	Dynamic bool `json:"dynamic"`
 
-	// Controller1Address is the controller-1 IPv4 or IPv6 network address value.
-	Controller1Address string `json:"controller1Address,omitempty"`
-
-	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=128
-	Prefix int `json:"prefix"`
-
-	// Gateway defines the nexthop gateway IP address if applicable
-	// +optional
-	Gateway *string `json:"gateway,omitempty"`
-
-	// Allocation defines the allocation scheme details for the network
-	Allocation AllocationInfo `json:"allocation"`
+	// AssociatedNetworks defines the set of network names to associate
```

Comment:
AssociatedNetworks or AssociatedAdressPools?

Reply:
Good catch!
It is "AssociatedAddressPools", updated the doc as well.